cmd/deploy/deploy.go

```diff
@@ -606,3 +591,35 @@ func (dc *DeployCommand) recreateFailedPods(ctx context.Context, name string) er
 	}
 	return nil
 }
+
+func (dc *DeployCommand) trackDeploy(manifest *model.Manifest, runInRemoteFlag bool, startTime time.Time, err error) {
+	deployType := "custom"
+	hasDependencySection := false
+	hasBuildSection := false
+	isRunningOnRemoteDeployer := false
+	if manifest != nil {
+		if manifest.IsV2 &&
+			manifest.Deploy != nil {
+			isRunningOnRemoteDeployer = isRemoteDeployer(runInRemoteFlag, manifest.Deploy.Image)
+			if manifest.Deploy.ComposeSection != nil &&
+				manifest.Deploy.ComposeSection.ComposesInfo != nil {
+				deployType = "compose"
+			}
+		}
+
+		hasDependencySection = manifest.HasDependencies()
+		hasBuildSection = manifest.HasBuildSection()
+	}
+
+	dc.AnalyticsTracker.TrackDeploy(analytics.DeployMetadata{
+		Success:                err == nil,
+		IsOktetoRepo:           utils.IsOktetoRepo(),
+		Duration:               time.Since(startTime),
+		PipelineType:           dc.PipelineType,
+		DeployType:             deployType,
+		IsPreview:              os.Getenv(model.OktetoCurrentDeployBelongsToPreview) == "true",
+		HasDependenciesSection: hasDependencySection,
```

Comment:
we can call directly to the `manifest.HasDependencies()` here and remove the var being declared, also for the BuildSection one. Could we make a helper method from manifest, maybe `IsCompose` to get the deployType same way? then we can remove all the nested ifs from here and delegate that to the manifest to give us that information? WDYT?

Reply:
I don't think so, what if it fails before manifest is already loaded it will fail because manifest will be nil